snyk/cli
# feat: support base64 encoding

#### What does this PR do?
- Adds support for base64 encoded payloads when executing `snyk code test`.
- Depends on `code-client` `4.11.0` (https://github.com/snyk/code-client/pull/144)

#### Where should the reviewer start?
- Review the above PR (https://github.com/snyk/code-client/pull/144). Changes made here are to support an additional boolean (`base64Encoding`).

#### How should this be manually tested?
- Ensure `code-client` `4.11.0` is available (either via `npm` after the above PR is merged, or by pulling the linked PR above and importing locally)
- A version of `deeproxy` with base64 encoding support is needed (https://github.com/snyk/deeproxy/pull/133)
- Set the `SNYK_USE_BASE64_ENCODING` environment variable to `true`
- Execute `snyk code test`

#### Any background context you want to provide?
This change is currently required for single tenant code analysis only.

Single tenant `deeproxy` instances are protected by Akamai WAF. When executing `snyk code test` the payload (source code transmitted in the clear) creates false positives and is blocked. Encoding to base64 should mask this content without need for more costly encryption activities.

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/NEBULA-520

#### Screenshots
-

#### Additional questions
-
